Modify content (Read Example+Getting Started) (#529)
heremaps/here-olp-sdk-cpp: docs/dataservice-read-catalog-example.md

   For instructions on how to get the access key ID and access key secret, see the [Get Credentials](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) section in the Terms and Permissions User Guide.

2. In `examples/dataservice-read/example.cpp`, replace the placeholders with your access key ID and access key secret.

   ```cpp
   const std::string kKeyId(""); // your here.access.key.id
   const std::string kKeySecret(""); // your here.access.key.secret
   ```

## Build and Run on Linux

To build and run the example on Linux:

1. Enable examples of the `CMake` targets.

   ```bash
   mkdir build && cd build
   cmake -DOLP_SDK_BUILD_EXAMPLES=ON ..
   ```

2. In the **build** folder, build the example.

   ```bash
   cmake --build . --target dataservice-read-example
   ```

3. Execute the example.

   ```bash
   ./examples/dataservice-read/dataservice-read-example
   ```

After building and running the example, you see the following information:

- `edge-example-catalog` &ndash; a catalog description
- `versioned-world-layer` &ndash; a layer description

> - `Request partition data - Success, data size - 3375` - a success message with an amount of data size

Reviewer: ```suggestion
- `Request partition data - Success, data size - 3375` - a success message which displays the size of retrieved data from partition
```
I think better be more explicit about the data size message

Author: Fixed

[INFO] read-example - Request partition data - Success, data size - 3375
```

## Build and Run on Android

> To integrate the HERE OLP SDK for C++ libraries in the Android project:

Reviewer: ```suggestion
To integrate the HERE OLP SDK for C++ libraries in the Android example project:
```

Author: Fixed

- [Set up prerequisites](#prerequisites-android)
- [Build the HERE OLP SDK for C++](#build-sdk-android)
- [Build and Run the APK](#build-and-run-android)

### <a name="prerequisites-android"></a>Prerequisites

> Before you integrate the HERE OLP SDK for C++ libraries in your Android project:

Reviewer: ```suggestion
Before you integrate the HERE OLP SDK for C++ libraries in Android example project:
```
This section describes how to use our Android example project, not custom user's project

Author: Thank you!
Fixed

   For more details, see [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).

   ```bash
   mkdir build && cd build
   cmake .. -DCMAKE_TOOLCHAIN_FILE=$NDK_ROOT/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DOLP_SDK_BUILD_EXAMPLES=ON -DOLP_SDK_ENABLE_TESTING=OFF
   ```

   The `CMake` command generates a `Gradle` project in the `build/examples/dataservice-read/android` folder.

5. Install the HERE OLP SDK for C++ libraries into the **sysroot** directory.

   ```bash
   # If necessary, execute as sudo.
   (sudo) make install
   ```

### <a name="build-and-run-android"></a>Build and Run the APK

To build and run the APK:

1. In the Android Studio IDE, open the `build/examples/dataservice-read/android/build.gradle` script.
2. Provide your application access key ID and access key secret.
3. Install and run the `dataservice_read_example` APK.

The main screen displays the following message: "Reading the partition data from the specified catalog completed successfully."

> If you encounter an error, in your device LogCat, find the error message.

Reviewer: ```suggestion
If you encounter an error, find the error message in your device Logcat.
```
The error doesn't happen in the Logcat, but in the application itself. Logcat just helps to debug this error message. As an alternative, we can get rid of this sentence.

Also, Locat, I guess, is not written in camel-case: https://developer.android.com/studio/command-line/logcat

Author: In this sentence, "in your device LogCat" indicates the place where the user should perform the action "find the error message". It doesn't denote the place where the error occurs as there is a comma in front of it. I would leave the sentence structure as is because the place of the action should go before the action (in tech writing).

Author: Deleted the sentence

## Build and run on iOS

> To integrate the HERE OLP SDK for C++ libraries in a basic iOS application written in the Objective-C language:

Reviewer: ```suggestion
To integrate the HERE OLP SDK for C++ libraries in a iOS example application written in the Objective-C language:
```

Author: Fixed

   For information on dependencies and installation instructions, see the [related section](https://github.com/heremaps/here-olp-sdk-cpp#dependencies) in the README.md file.
3. Replace the placeholders in `examples/dataservice-read/example.cpp` with your application access key ID and access key secret.
   For instructions on how to get the access key ID and access key secret, see the [Get Credentials](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) section in the Terms and Permissions User Guide.

### <a name="build-sdk-ios"></a>Build the HERE OLP SDK for C++

To build the HERE OLP SDK for C++ on iOS:

1. Set `OLP_SDK_BUILD_EXAMPLES` to `ON`.
2. (Optional) To disable tests, set `OLP_SDK_ENABLE_TESTING` to `OFF`.
3. Specify the path to the iOS toolchain file using the `CMAKE_TOOLCHAIN_FILE` variable.
   > **Note:** The iOS toolchain file is shipped together with the SDK and located under the `<olp-sdk-root>/cmake/toolchains/iOS.cmake`.

```bash
mkdir build && cd build
cmake .. -GXcode  -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/iOS.cmake -DPLATFORM=iphoneos -DOLP_SDK_BUILD_EXAMPLES=ON -DOLP_SDK_ENABLE_TESTING=OFF
```

> To configure the HERE OLP SDK for C++ for a simulator, set the `SIMULATOR` variable to `ON`.

Reviewer: ```suggestion
To configure the HERE OLP SDK for C++ for a simulator, set the `SIMULATOR` CMake variable to `ON`.
```

Author: Fixed

### <a name="build-and-run-ios"></a>Build and Run the Application

> To build an run the application on iOS:

Reviewer: ```suggestion
To build an run the example application on iOS:
```

Author: Fixed
